File: 04_agent/strands-python/skills/summarize.py
from tools.task_tools import task_store


def summarize() -> str:
    total = len(task_store)
    done = sum(1 for t in task_store if t["status"] == "done")
    in_progress = sum(1 for t in task_store if t["status"] == "in_progress")
    todo = sum(1 for t in task_store if t["status"] == "todo")
    return f"タスク合計: {total}件（完了: {done}件、進行中: {in_progress}件、未着手: {todo}件）"

File: 04_agent/mastra/backend/src/skills/summarize.ts
import { listTasks } from "../tools/taskTools.js";

export const summarize = (): string => {
  const tasks = listTasks();
  const done = tasks.filter((t) => t.status === "done").length;
  const inProgress = tasks.filter((t) => t.status === "in_progress").length;
  const todo = tasks.filter((t) => t.status === "todo").length;
  return `タスク合計: ${tasks.length}件（完了: ${done}件、進行中: ${inProgress}件、未着手: ${todo}件）`;
};
